internal/jujuapi/jimm.go

```diff
@@ -689,3 +691,38 @@ func (r *controllerRoot) StartBootstrapJob(ctx context.Context, req apiparams.Bo
 		JobID: jobID,
 	}, nil
 }
+
+// StartDestroyControllerJob starts a destroy-controller job
+func (r *controllerRoot) StartDestroyControllerJob(ctx context.Context, req apiparams.DestroyControllerRequest) (apiparams.StartJobResponse, error) {
```

Comment:
I don't see a test for this, at least verifying our logic for only destroying controllers with 0 models.